src/Auth/CanDirective.php

```diff
@@ -124,19 +124,18 @@ public function handleField(FieldValue $fieldValue, \Closure $next): FieldValue
         $resolved = $this->directiveArgValue('resolved');
 
         $fieldValue->setResolver(function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($previousResolver, $ability, $resolved) {
-            $gate = $this->gate->forUser($context->user());
```

Comment:
I am unsure if removing this will have side effects. Let's investigate when and why it was added.

Reply:
That line was added in PR #1043 and it was not addressed why.
But I think it wouldn't have side effects given `Gate` uses the config's default guard as `Nuwave\Lighthouse\Schema\Context`, the difference is `Context` resolves the user before `@guard` and `Gate` resolves the user until it tries to authorize.
Another solution could be that `Context` resolves the user until `user()` is called, so if it's called after `@guard` the default guard changes and doesn't return null.